consensus/src/state_computer.rs

```diff
@@ -333,6 +349,17 @@ impl StateComputer for ExecutionProxy {
             LogicalTime::new(target.ledger_info().epoch(), target.ledger_info().round());
         let block_timestamp = target.commit_info().timestamp_usecs();
 
+        if self.executor.committed_block_id() == target.commit_info().id() {
+            return Ok(());
+        }
+        if self.executor.latest_synced_version() == target.commit_info().version() {
```

Comment:
(should distinguish between synced version and pre-committed version, and..)
shall we check the state hashes match?

Reply:
isn't the hash checked inside the commit_ledger?